### PR TITLE
#445 - Make ResourceAssemblerSupport extensible

### PR DESCRIPTION
**Pull request by gregturn**
Adds getters to allow implementers access to controllerClass and resourceType.

### CODE REVIEW

File: src/main/java/org/springframework/hateoas/mvc/ResourceAssemblerSupport.java [hunk 1]
```diff
@@ -58,7 +60,7 @@ public ResourceAssemblerSupport(Class<?> controllerClass, Class<D> resourceType)
 	 * @param entities must not be {@literal null}.
 	 * @return
 	 */
-	public List<D> toResources(Iterable<? extends T> entities) {
+	public Resources<D> toResources(Iterable<? extends T> entities) {
```
Comment (schauder) [inline src/main/java/org/springframework/hateoas/mvc/ResourceAssemblerSupport.java]: This clashes with /duplicates  #416 and seems unrelated to this issue.

File: src/test/java/org/springframework/hateoas/mvc/ResourceAssemblerSupportTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	@Test
+	public void resourceAssemblerShouldConvertCollectionOfEntitiesIntoResources() {
+
+		EmployeeAssembler assembler = new EmployeeAssembler();
+		Employee employee1 = new Employee("Frodo");
+		Employee employee2 = new Employee("Bilbo");
+
+		Resources<EmployeeResource> employees = assembler.toResources(Arrays.asList(employee1, employee2));
+
+		assertThat(employees, is(notNullValue()));
```
Comment (schauder) [inline src/test/java/org/springframework/hateoas/mvc/ResourceAssemblerSupportTest.java]: I think we should we use AssertJ instead of Hamcrest.

File: src/test/java/org/springframework/hateoas/mvc/ResourceAssemblerSupportTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import org.springframework.hateoas.Resources;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+/**
+ * @author Greg Turnquist
+ */
+public class ResourceAssemblerSupportTest {
+
+	@Test
```
Comment (schauder) [inline src/test/java/org/springframework/hateoas/mvc/ResourceAssemblerSupportTest.java]: Comments referencing the issue are missing.

**Comment by gregturn** [comment]
Superceded by pull-request #1009.
